packages/ms-authenticator/src/authenticators/MsAuthenticator.ts

```diff
@@ -0,0 +1,126 @@
+import { ConfidentialClientApplication, LogLevel, PublicClientApplication, UsernamePasswordRequest } from '@azure/msal-node'
+import {
+  IMsAuthenticationAuthorizationCodeArgs,
+  IMsAuthenticationClientCredentialArgs,
+  IMsAuthenticationOnBehalfOfArgs, IMsAuthenticationSilentFlowArgs,
+  IMsAuthenticationUsernamePasswordArgs,
+} from '../index'
+
+import { fetch } from 'cross-fetch'
+
+/**
+ * Not implemented yet
+ * @param authenticationArgs
+ * @constructor
+ */
+export async function AuthorizationCodeAuthenticator(authenticationArgs: IMsAuthenticationAuthorizationCodeArgs): Promise<string> {
+  throw new Error("This authentication method is not implemented yet.")
+}
+
+/**
+ * Not implemented yet
+ * @param authenticationArgs
+ * @constructor
+ */
+export async function BehalfOfAuthenticator(authenticationArgs: IMsAuthenticationOnBehalfOfArgs): Promise<string> {
+  throw new Error("This authentication method is not implemented yet.")
+}
+
+/**
+ * necessary fields are:
+ *   azClientId: clientId of the application you're trying to login
+ *   azClientSecret: secret of the application you're trying to login
+ *   azTenantId: your MS Azure tenantId
+ *   credentialManifest: address of your credential manifest. usually in following format:
+ *    https://beta.eu.did.msidentity.com/v1.0/<tenant_id>/verifiableCredential/contracts/<verifiable_credential_schema>
+ * @param authenticationArgs
+ * @constructor
+ */
+export async function ClientCredentialAuthenticator(authenticationArgs: IMsAuthenticationClientCredentialArgs): Promise<string> {
+  var msalConfig = {
+    auth: {
+      clientId: authenticationArgs.azClientId,
+      authority: 'https://login.microsoftonline.com/' + authenticationArgs.azTenantId,
```

Comment:
Are we sure this always has this form?

Reply:
change to read from an optional field, if not, this will be the format